webauthn-rs-core/src/core.rs

```diff
@@ -163,21 +126,19 @@ impl WebauthnCore {
     /// persist. It is strongly advised you associate this RegistrationState with the
     /// UserId of the requester.
     #[allow(clippy::too_many_arguments)]
-    pub fn generate_challenge_register_options(
+    pub fn generate_challenge_register(
```

Comment:
Replace options with a struct with a builder pattern. That will let us encode the defaults which existed on the old `generate_challenge_register()` function.